src/iris/bin/sync_targets.py

```diff
@@ -391,8 +391,7 @@ def sync_from_oncall(config, engine, purge_old_users=True):
             except SQLAlchemyError as e:
                 logger.exception('Error inserting oncall_team %s: %s', t, e)
                 continue
-
-    session.commit()
+        session.commit()
```

Comment:
Whoah-- this function has ~400 LoC. That needs to be addressed before doing anything else. I defy you to explain to me the impact of this change for every code path through this method.

Reply:
Reverted since this would be unnecessary as the inserts in this function directly use engine.execute() so the session doesn't actually have anything to commit. I completely agree that this should be refactored into smaller functions though perhaps that is best left to a different pull request. I think taking a second look at how we use sqlalchemy, if not replace it entirely, is in the horizon but again a bit out of scope for this change.